runtime/queries/metricsview_aggregation.go

```diff
@@ -576,6 +592,18 @@ func (q *MetricsViewAggregation) buildMetricsAggregationSQL(mv *runtimev1.Metric
 	return sql, args, nil
 }
 
+func applyFilter(mv *runtimev1.MetricsViewSpec, expr string, filter *runtimev1.Expression, args *[]any, dialect drivers.Dialect) (string, error) {
+	if filter != nil {
+		condClause, condArgs, err := buildConditionExpression(mv, filter.GetCond(), nil, dialect)
```

Comment:
Should use `buildExpression` on `filter` here since `filter` is not guaranteed to be a `Cond` any more